dev/git.ts

```diff
@@ -200,6 +200,53 @@ const gitGenerators: Record<string, Fig.Generator> = {
       });
     },
   },
+
+  // Wildcard
+  wildcard_files_for_staging: {
+    script: "git status --short",
+    postProcess: (out) => {
+      const output = filterMessages(out);
+      if (output.startsWith("fatal:")) {
+        return [];
+      }
+
+      const items = output.split("\n").map((file) => {
+        file = file.slice(0, file.lastIndexOf("/") + 1);
+        file = file.trim();
+        const arr = file.split(" ");
+        return arr.slice(1).join(" ").trim();
+      });
+
+      if (items.length < 2) {
+        return [];
+      }
+
+      var dirArr = [];
+      var currentDir = items[0];
+      var count = 1;
+      for (var i = 1; i < items.length; i++) {
+        if (items[i].includes(currentDir) && i + 1 !== items.length) {
+          currentDir = items[i];
+          count++;
+        } else if (count >= 2) {
+          dirArr.push(currentDir);
+          currentDir = items[i];
+          count = 1;
+        } else {
+          currentDir = items[i];
+          count = 1;
+        }
```

Comment:
We can simplify this a bit because some parts are duplicated, something like this (needs to be tested):
```suggestion
        currentDir = items[i];
        
        if (items[i].includes(currentDir) && i + 1 !== items.length) {
          count++;
        } else {
          if (count >= 2) {
            dirArr.push(currentDir);
          }
          
          count = 1;
        }
```